src/sparse/KokkosSparse_spmv.hpp

```diff
@@ -344,13 +344,11 @@ spmv (KokkosKernels::Experimental::Controls controls,
   // Call single-vector version if appropriate
   if (x.extent(1) == 1) {
     typedef Kokkos::View<typename XVector::const_value_type*,
-      typename Kokkos::Impl::if_c<std::is_same<typename YVector::array_layout, Kokkos::LayoutLeft>::value,
-                                  Kokkos::LayoutLeft, Kokkos::LayoutStride>::type,
+      typename YVector::array_layout,
```

Comment:
I think this won't work if X is a single column but noncontiguous (like a column of a LayoutRight view), and Y is contiguous. This should use the UnifiedLayout.